nomos-cli/src/cmds/disseminate/mod.rs

```diff
@@ -0,0 +1,217 @@
+use nomos_network::{backends::libp2p::Libp2p, NetworkService};
+use overwatch_derive::*;
+use overwatch_rs::{
+    overwatch::OverwatchRunner,
+    services::{
+        handle::{ServiceHandle, ServiceStateHandle},
+        relay::NoMessage,
+        state::*,
+        ServiceCore, ServiceData, ServiceId,
+    },
+    DynError,
+};
+use std::{path::PathBuf, time::Duration};
+
+#[derive(Args, Debug)]
+pub struct Disseminate {
+    // TODO: accept bytes
+    #[clap(short, long)]
+    data: String,
+    /// Path to the network config file
+    #[clap(short, long)]
+    network_config: PathBuf,
+    /// The data availability protocol to use. Defaults to full replication.
+    #[clap(flatten)]
+    da_protocol: DaProtocolChoice,
+    /// Timeout in seconds. Defaults to 120 seconds.
+    #[clap(short, long, default_value = "120")]
+    timeout: u64,
+}
+
+#[derive(Debug, Clone, Args)]
+pub struct FullReplicationSettings {
+    #[clap(long, default_value = "1")]
+    num_attestations: usize,
+}
+
+impl Disseminate {
+    pub fn run(&self) -> Result<(), Box<dyn std::error::Error>> {
+        tracing::subscriber::set_global_default(tracing_subscriber::FmtSubscriber::new())
+            .expect("setting tracing default failed");
+        let network = serde_yaml::from_reader::<
+            _,
+            <NetworkService<Libp2p> as ServiceData>::Settings,
+        >(std::fs::File::open(&self.network_config)?)?;
+        OverwatchRunner::<DisseminateApp>::run(
+            DisseminateAppServiceSettings {
+                network,
+                send_blob: Settings {
+                    bytes: self.data.clone().as_bytes().into(),
+                    timeout: Duration::from_secs(self.timeout),
+                    da_protocol: self.da_protocol.clone(),
+                },
+            },
+            None,
+        )
+        .unwrap()
+        .wait_finished();
+        Ok(())
+    }
+}
+
+// Write '✓' at the end of the previous line in terminal
+fn terminal_cmd_done() {
+    let go_to_previous_line = crossterm::cursor::MoveToPreviousLine(1);
+    let go_to_end_of_line =
+        crossterm::cursor::MoveToColumn(crossterm::terminal::size().unwrap().0 - 1);
+    let write_done = crossterm::style::Print("✓");
+    execute!(
+        std::io::stdout(),
+        go_to_previous_line,
+        go_to_end_of_line,
+        write_done
+    )
+    .unwrap()
+}
+
+async fn disseminate<D, B, N, A, C>(
+    mut da: D,
+    data: Box<[u8]>,
+    adapter: N,
+) -> Result<C, Box<dyn std::error::Error>>
+where
+    D: DaProtocol<Blob = B, Attestation = A, Certificate = C>,
+    N: NetworkAdapter<Blob = B, Attestation = A> + Send + Sync,
+{
+    // 1) Building blob
+    tracing::info!("Building blobs...");
+    let blobs = da.encode(data);
+    terminal_cmd_done();
+
+    // 2) Send blob to network
+    tracing::info!("Sending blobs to network...");
+    for blob in blobs {
+        adapter.send_blob(blob).await.unwrap();
+    }
```

Comment:
We can send then all at once (they may be a lot of chunks eventually), something like:

```rust
let tasks: FuturesUnordered<_> = blobs.map(|blob| tokio::spawnadapter.send_blob(blob))).collect()
future::try_join_all(tasks).await;
```